### PR TITLE
Feat: Update comment endpoint

### PR DESCRIPTION
**Pull request by kihuni**
## Description

This pull request introduces a new PUT endpoint /api/v1/comments/{comment_id}/ to update existing comments. The endpoint allows users to update the content of a specific comment by its ID.

​
## Related Issue [Link to issue ticket](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/518)

Motivation and Context
This change is required to allow users to modify their comments after they have been posted. This solves the issue of having incorrect or outdated information in comments.

- Tested the PUT endpoint using Postman to ensure it updates comments correctly.
- Created unit tests to verify the endpoint behavior with valid and invalid inputs.
- Ran integration tests to ensure the endpoint works as expected within the application flow.

Types of changes
 

- [ ] Bug fix (non-breaking change which fixes an issue)

 

- [x] New feature (non-breaking change which adds functionality)

 

- [ ] Breaking change (fix or feature that would cause existing functionality to change)

Checklist

 

- [x] My code follows the code style of this project.

 

- [x] My change requires a change to the documentation.

- [x]  I have updated the documentation accordingly.

- [x]  I have read the CONTRIBUTING document.

 

- [x] I have added tests to cover my changes.

 

- [x] All new and existing tests passed.

### CODE REVIEW

File: api/v1/services/comment.py
Comment (joboy-dev) [inline api/v1/services/comment.py:1]: What is with the removed code??

**Review by joboy-dev — COMMENTED** [review]
What is with the removed code??

**Comment by chukssomzzy** [comment]
I have  worked on this, I have been waiting to be merged for close to 6 days now. just getting moved around...

**Comment by chukssomzzy** [comment]
PR 455

**Comment by chukssomzzy** [comment]
You saw a pr is open and is active and you still went ahead to implement the feature that not cool bro.

**Pull request closed without merging**